File: src/segundo-bimestre/llp/s14/banco-dielme/src/models/Usuario.py
class Usuario:
    def __init__(self, nome, email, senha):
        self.__codigo_usuario = 0
        self.setCodigoUsuario()
        self.__nome = nome
        self.__email = email
        self.__senha = senha

    def getCodigoUsuario(self):
        return self.__codigo_usuario

    def setCodigoUsuario(self):
        if self.__codigo_usuario == 0:
            self.__codigo_usuario += 1
        else:
            self.__codigo_usuario += 1
        
    def getNome(self):
        return self.__nome

    def setNome(self, nome):
        self.__nome = nome
        
    def getEmail(self):
        return self.__email
    
    def setEmail(self, email):
        self.__email = email
        
    def getSenha(self):
        return self.__senha
    
    def setSenha(self, senha):
        self.__senha = senha
        
    def mostrar_usuario(self):
        return f"Código: {self.getCodigoUsuario()}, Nome: {self.getNome()}, Email: {self.getEmail()}"

File: src/quarto-bimestre/back-end/teste-ts/Usuario.js
"use strict";
exports.__esModule = true;
var Usuario = /** @class */ (function () {
    function Usuario(id, nome, email, senha, tipoUsuario) {
        this.id = id;
        this.email = email;
        this.nome = nome;
        this.senha = senha;
        this.tipoUsuario = tipoUsuario;
    }
    Usuario.prototype.getId = function () {
        return this.id;
    };
    Usuario.prototype.getNome = function () {
        return this.nome;
    };
    Usuario.prototype.getEmail = function () {
        return this.email;
    };
    Usuario.prototype.getTipoUsuario = function () {
        return this.tipoUsuario;
    };
    Usuario.prototype.getSenha = function (tipoUsuario) {
        if (tipoUsuario === 1) {
            return this.senha;
        }
        else {
            console.log("\nA senha s\u00F3 pode ser mostrada a um usu\u00E1rio que \u00E9 Administrador");
        }
    };
    Usuario.prototype.exibirDetalhes = function (tipoUsuario) {
        console.log("\nID: " + this.getId());
        console.log("Nome: " + this.getNome());
        console.log("Email: " + this.getEmail());
        console.log("Senha: " + this.getSenha(tipoUsuario));
        console.log("TipoUsuario: " + this.getTipoUsuario() + "\n");
    };
    return Usuario;
}());
exports["default"] = Usuario;
